### PR TITLE
Handle curl exceptions and timeouts better (Issue #693)

### PR DESCRIPTION
**Pull request by abraunegg**
* When curl experiences a timeout, exit the application rather than send back an empty JSON for handling and sending the application into an endless loop

### CODE REVIEW

**Comment by abraunegg** [comment]
@norbusan 
Whilst this now exits, if using systemd, this now would rely on the systemd file to restart the application:
```
[Unit]
Description=OneDrive Free Client
Documentation=https://github.com/abraunegg/onedrive
After=network-online.target
Wants=network-online.target

[Service]
ExecStart=/usr/local/bin/onedrive --monitor
Restart=on-failure
RestartSec=3
RestartPreventExitStatus=3

[Install]
WantedBy=default.target
```

On application restart however, if the Internet is still not available, it will gracefully fail again until it is available.

**Comment by abraunegg** [comment]
@norbusan 
To test this:

**block https**
```
firewall-cmd --direct --add-rule ipv4 filter OUTPUT 0 -p tcp -m tcp --dport=443 -j DROP
firewall-cmd --reload
iptables -L -n | grep 443
```

**unblock https**
```
service firewalld restart
iptables -L -n | grep 443
```

**Comment by abraunegg** [comment]
@norbusan 
**without patch**
```
Uploading new items of .
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: 66D53BE8A5056ECA!101
ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
ERROR: onedrive.getPathDetailsById call returned an invalid JSON Object
ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
ERROR: onedrive.viewChangesByItemId call returned an invalid JSON Object
ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
ERROR: onedrive.viewChangesByItemId call returned an invalid JSON Object
ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
ERROR: onedrive.viewChangesByItemId call returned an invalid JSON Object
ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
ERROR: onedrive.viewChangesByItemId call returned an invalid JSON Object
^CGot termination signal, shutting down db connection
```

**with patch**
```
Uploading new items of .
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: 66D53BE8A5056ECA!101
Syncing changes from OneDrive ...
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: 66D53BE8A5056ECA!101
ERROR: OneDrive returned an error with the following message:
  Error Message: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
Exiting application due to Internet connectivity issues
```

File: src/onedrive.d [hunk 1]
```diff
@@ -629,8 +630,22 @@ final class OneDriveApi
...
+			log.error("ERROR: OneDrive returned an error with the following message:");
+			auto errorArray = splitLines(e.msg);
+			string errorMessage = errorArray[0];
+			
+			if (canFind(errorMessage, "Couldn't connect to server on handle")) {
+				// This is a curl timeout
+				log.error("  Error Message: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?");
+				// Must exit here
+				log.error("Exiting application due to Internet connectivity issues");
+                exit(-1);
```
Comment (norbusan) [inline src/onedrive.d]: Spacing seems to be strange here

Comment (abraunegg) [reply src/onedrive.d]: Thats weird as I have checked the file locally & the spacing is 100% fine ... odd ... replacing with 'spaces'

Comment (abraunegg) [reply src/onedrive.d]: If you look at the file online the spacing is right .. Microsoft / Github must have changed something in the API - as the spacing is right ??

https://github.com/abraunegg/onedrive/blob/Issue-%23693/src/onedrive.d

**Comment by abraunegg** [comment]
CLosing issue as PR #695 is a cleaner solution

**Pull request closed without merging**